docker-compose.override.yml

```diff
@@ -13,6 +13,7 @@ services:
     environment:
       - REACT_APP_NAME=webapp-dev
       - PORT=80
+      - WDS_SOCKET_PORT=0
```

Comment:
I don't really get this change, could you elaborate a bit more?

From what I understand, If we use sockets, our webapp will connect to backend as a client, so no need to expose ports? Why would we have a problem with this?

Reply:
Sorry, I didn't understand either when I made the change :sweat_smile:. After some research, here is how I think it works. Please correct me if I am wrong.

[WDS_SOCKET_PORT](https://create-react-app.dev/docs/advanced-configuration/) is an env variable of the Webpack Dev Server(WDS) for setting a custom WebSocket port for hot module reloading.

Create React App(CRA) uses WDS in development mode.

Also, we are using different ports for reverse proxy and the React app in development, port 3000 for proxy and port 80 for the React app.
We use port 80 for the React app in development and production to have only one config for the reverse proxy and use it both for development and production.

CRA configures the WDS to use port 80 (the one we specify with the PORT env) and also sets port 80 for the custom WebSocket port for hot module reloading.

When the browser opens a WebSocket client connection on that port (80) for the hot reload, it does not work because the server is behind a proxy and the proxy uses port 3000.

The solution is to either set WDS_SOCKET_PORT to the same port as the reverse proxy (3000) or set it to 0 and let the WDS client figure it out from the window.location.port(from the browser address), which in our case, for development, is 3000.

For the production environment, we use nginx, which is why we add WDS_SOCKET_PORT in this docker-compose file only for development.

I hope it makes sense.

Reply:
Should we set this env to 0 or 3000?

Reply:
Thanks for the explanation, it makes a lot of sense now :+1: For mi `0` is a better fit just in case we decide to update reserse-proxy ports in the future for whatever reason